app/utils.py

```diff
@@ -152,25 +152,44 @@ def setup_logger(name, level=logging.INFO):
     return logger
 
 
-def validate_resource(json):
+def validate_resource(request, id=-1):
+    json = request.get_json()
     validation_errors = {'errors': {'type': 'validation'}}
+    missing_params = []
+    invalid_params = []
+
     if not json:
         message = "A JSON body is required to use this endpoint, but none was given"
         validation_errors['errors']['message'] = message
         return validation_errors
 
-    validation_errors['errors']['missing_params'] = []
-
-    required = []
-    for column in Resource.__table__.columns:
-        if column.nullable is False and column.name != 'id':
-            # strip _id from category_id
-            name = column.name.replace('_id', '')
-            required.append(name)
-
-    for prop in required:
-        if json.get(prop) is None:
-            validation_errors['errors']['missing_params'].append(prop)
-
-    if validation_errors['errors']['missing_params']:
+    # There are only required parameters for creating new Resources
+    if request.method == 'POST':
+        required = []
+        for column in Resource.__table__.columns:
+            if column.nullable is False and column.name != 'id':
+                # strip _id from category_id
+                name = column.name.replace('_id', '')
+                required.append(name)
+
+        for prop in required:
+            if json.get(prop) is None:
+                missing_params.append(prop)
+
+    url = json.get("url")
+    if url is not None:
+        # If there is an existing resource with the requested url and it isn't the Resource we are trying to update.
+        resource = Resource.query.filter_by(url=url).first()
+        if resource and resource.id != id:
+            invalid_params.append("url")
```

Comment:
We should have a message in `validation_errors['errors']['message']` that states the url already exists in the database. Otherwise, people will be wondering why it's invalid. And since we're already pulling the resource, we have the id. So we can provide that as well. You'd get a response like this:
```json
{
    "apiVersion": "1.0",
    "data": null,
    "errors": {
        "invalid_params": [
            "url"
        ],
         "message": "This URL already exists in the database. Resource id 135 already has this URL.",
        "type": "validation"
    },
    "status": "Unprocessable Entity",
    "status_code": 422
}
```

Reply:
In accordance with my latest comment regarding the `code` attribute, I'm updating the above example to include that as well